src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs

```diff
@@ -0,0 +1,511 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using WindowsInput.Native;
+using Xunit;
+using Xunit.Abstractions;
+using static Interop.UiaCore;
+
+namespace System.Windows.Forms.UITests
+{
+    // This class contains many tests which don't require user input.
+    // Although it looks like they should be moved to the unit tests project, it doesn't.
+    // Unit tests don't support the "Tile" mode, so all those tests belong here.
+
+    public class ListViewTests : ControlTestBase
+    {
+        public ListViewTests(ITestOutputHelper testOutputHelper)
+            : base(testOutputHelper)
+        {
+        }
+
+        [WinFormsTheory]
+        [InlineData(false, true, true)]
+        [InlineData(false, false, false)]
+        [InlineData(true, true, false)]
+        public async Task ListView_CheckBox_DoubleClick_WorksExpectedAsync(bool virtualModeEnabled, bool checkBoxesEnabled, bool expected)
+        {
+            await RunTestAsync(async (form, listView) =>
+            {
+                InitializeItems(listView, View.Details, virtualModeEnabled, checkBoxesEnabled);
+                Point itemCenter = GetCenter(listView.RectangleToScreen(listView.Items[0].Bounds));
+
+                await MoveMouseAsync(form, itemCenter);
+
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Mouse.LeftButtonDoubleClick());
+                Assert.Equal(expected, listView.Items[0].Checked);
```

Comment:
A typical test consists of three "A"s:
1. Arrange
2. Act
3. Assert

And it is helpful when these three "A"s are visually separated:
```suggestion
                InitializeItems(listView, View.Details, virtualModeEnabled, checkBoxesEnabled);

                Point itemCenter = GetCenter(listView.RectangleToScreen(listView.Items[0].Bounds));
                await MoveMouseAsync(form, itemCenter);
                await InputSimulator.SendAsync(
                    form,
                    inputSimulator => inputSimulator.Mouse.LeftButtonDoubleClick());

                Assert.Equal(expected, listView.Items[0].Checked);
```

Ditto below

Reply:
@Tanya-Solyanik On what test style should I focus?
https://github.com/dotnet/winforms/pull/6257#discussion_r762255582